FBTweak/FBTweak.m

```diff
@@ -179,17 +181,39 @@ - (instancetype)initWithIdentifier:(NSString *)identifier name:(NSString *)name
                       defaultValue:(FBTweakValue)defaultValue {
   if (self = [super initWithIdentifier:identifier name:name defaultValue:defaultValue]) {
     NSData *archivedValue = [[NSUserDefaults standardUserDefaults] objectForKey:identifier];
-    self.currentValue = (archivedValue != nil && [archivedValue isKindOfClass:[NSData class]] ?
-                         [NSKeyedUnarchiver unarchiveObjectWithData:archivedValue] : archivedValue);
+    if (archivedValue != nil && [archivedValue isKindOfClass:[NSData class]]) {
+      self.currentValue =
+          [NSKeyedUnarchiver unarchivedObjectOfClasses:[FBPersistentTweak validValueClasses]
+                                              fromData:archivedValue error:nil];
+    } else {
+      self.currentValue = archivedValue;
+    }
   }
 
   return self;
 }
 
 - (void)setCurrentValue:(FBTweakValue)currentValue {
   [super setCurrentValue:currentValue];
+
   // we can't store UIColor to the plist file. That is why we archive value to the NSData.
-  [[NSUserDefaults standardUserDefaults] setObject:[NSKeyedArchiver archivedDataWithRootObject:currentValue] forKey:self.identifier];
+  NSError *error;
+  NSData *data = [NSKeyedArchiver archivedDataWithRootObject:currentValue requiringSecureCoding:YES
+                                                   error:&error];
+  NSAssert(data != nil, @"Failed to archive value with error: %@", error);
```

Comment:
add the type of the value. I'd add a more "user friendly" msg like "Can't use type \(type) with tweaks" or something

Reply:
Well, the type can be for example an array that contains a type that can't be archived, and then the message you suggest will be misleading. I think it is best to print the archiving error as it will give the most accurate information.